addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/InputWithPlayControl.java

```diff
@@ -0,0 +1,257 @@
+        public String Artist; // USB, iPOD, PC
+        public String Album; // USB, iPOD, PC
+        public String Song; // USB, iPOD, PC
+
+        public String playbackMode = "Stop"; // All inputs
+
+        public void invalidate() {
+            this.playbackMode = "N/A";
+            this.Station = "N/A";
+            this.Artist = "N/A";
+            this.Album = "N/A";
+            this.Song = "N/A";
+        }
+    }
+
+    public static class PlayControlState {
+        public int presetChannel = 0; // NET_RADIO, RADIO, HD_RADIO, iPOD, USB, PC
+
+        public void invalidate() {
+            presetChannel = 0;
+        }
+    }
+
+    public interface Listener {
+        void playInfoUpdated(PlayInfoState msg);
+
+        void playControlUpdated(PlayControlState msg);
+    }
+
+    private Listener observer;
+
+    /**
+     * Create a InputWithPlayControl object for altering menu positions and requesting current menu information as well
+     * as controlling the playback and choosing a preset item.
+     *
+     * @param inputID The input ID like USB or NET_RADIO.
+     * @param com The Yamaha communication object to send http requests.
+     */
+    public InputWithPlayControl(String inputID, HttpXMLSendReceive com, Listener observer) {
+        this.inputID = inputID;
+        this.com_ref = new WeakReference<HttpXMLSendReceive>(com);
+        this.observer = observer;
+    }
+
+    /**
+     * Wraps the XML message with the inputID tags. Example with inputID=NET_RADIO:
+     * <NETRADIO>message</NETRADIO>.
+     *
+     * @param message XML message
+     * @return
+     */
+    protected String wrInput(String message) {
+        return "<" + inputID + ">" + message + "</" + inputID + ">";
+    }
+
+    /**
+     * Updates the playback information
+     *
+     * @throws Exception
+     */
+    public void updatePlaybackInformation() throws IOException, ParserConfigurationException, SAXException {
+        if (observer == null) {
+            return;
+        }
+
+        HttpXMLSendReceive com = com_ref.get();
+        String response = com.post(wrInput("<Play_Info>GetParam</Play_Info>"));
+        Document doc = com.xml(response);
+        if (doc.getFirstChild() == null) {
+            throw new SAXException("<Play_Info>GetParam failed: " + response);
+        }
+
+        PlayInfoState msg = new PlayInfoState();
+
+        Node playbackInfoNode = HttpXMLSendReceive.getNode(doc.getFirstChild(), "Play_Info/Playback_Info");
+        msg.playbackMode = playbackInfoNode != null ? playbackInfoNode.getTextContent() : "";
+
+        Node metaInfoNode = HttpXMLSendReceive.getNode(doc.getFirstChild(), "Play_Info/Meta_Info");
+        Node sub;
+
+        if (inputID.equals("TUNER")) {
+            sub = HttpXMLSendReceive.getNode(metaInfoNode, "Radio_Text_A");
+            msg.Station = sub != null ? sub.getTextContent() : "";
```

Comment:
Can you elaborate on that? Look at the line above those checks, sub is a different xml node each time. I didn't want to clutter the code with a new variable each time and I am reusing the 'sub' variable instead.